### ISSUE TITLE
Support for "Decontam"

### ISSUE THREAD

**Issue by erikrikarddaniel** [nf-core/ampliseq#224]
The "Decontam" package (https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-018-0605-2), by the same author as DADA2, looks like an interesting way of getting rid of contaminants, especially when one doesn't have negative controls.

Would be good to have in ampliseq. This might be easier if we go for a pure R/DADA2 implementation.

**Comment by erikrikarddaniel** [comment]
Later, I realized this package is dependent on blanks sequenced with the real samples. Not a showstopper, but makes it a bit less interesting maybe.

**Comment by d4straub** [comment]
I suggested a few times in project planning to include blanks as negative control, but it was never done. On the other hand I do have a project were the microbiome inside a rock is sampled and a bunch of contamination sources are sequenced as well. But those are honestly easier to interpret by just looking at ASV tables (and the wet-lab scientist would never trust a program removing contaminants... they want to have the numbers of all samples to look at and then decide themselves). So I did not have a need for it in the past or currently.

**Comment by erikrikarddaniel** [comment]
I just spent two days analyzing samples with a strong batch effect (second PCA axis) that turned out to be due to likely contamination. Luckily, we had blanks sequenced, so keep suggesting. I've seen this in other projects as well where DNA concentrations were low and varying.

**Comment by hjarnek** [comment]
_"Right now, it is widespread practice to address contamination by sequencing negative controls, and then doing nothing"_ – Benjamin Callahan, 2018

I think sequencing negative controls at all can be considered a best-case scenario, as exemplified by this thread. But my impression is that this has been changing over the last few years. Reviewers are now more aware of the issue with blank correction and it is increasingly being requested that you consider it, that's what I hear at least. _decontam_ is AFAIK the only tailored piece of software out there that has some statistical basis to identifying contaminants. It's easy to use, reproducible, and more sophisticated than using static cutoffs or eyeballing it. I think it's time to re-open this issue, @d4straub.

**Comment by d4straub** [comment]
Yes, I agree that it would be nice to have decontam. If anyone is interested to add that to the pipeline, I would be happy to look at a PR. I think its appropriate to open it, but it can be also closedif you disagree @erikrikarddaniel .

**Comment by erikrikarddaniel** [comment]
I can only agree

**Comment by d4straub** [comment]
Thats in the dev branch now, closing this issue.
